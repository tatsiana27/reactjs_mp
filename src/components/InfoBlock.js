import React, { Component } from 'react';
import { RadioButton } from './RadioButton';
import { connect } from 'react-redux';
import { SORTINGS } from '../data/data';

import '../styles/info-block.scss';

export class InfoBlockUI extends Component {
    state = {
        sortBy: 'raiting'
    }
    onRadioChange = (event) => {
        this.setState({
            sortBy: event.currentTarget.value
        });
    };
    render () {
        const { sortBy } = this.state;
        const { movies } = this.props;

        return (
            <div className="info-block">
                <span className="movies-count">{movies.data.length} movies found</span>
                <div className="sorting-block">
                    <p>Sort by</p>
                    {SORTINGS.map(item =>
                        <RadioButton
                            key={item.id}
                            name={item.name}
                            id={item.id}
                            value={item.name}
                            checked={sortBy === item.name}
                            onChange={this.onRadioChange}
                        />)
                    }
                </div>
            </div>
        );
    }
}

export const mapStateToProps = ({movies}) => {
    return {
        movies: movies,
    };
};

export const mapDispatchToProps = (dispatch) => ({});

export const InfoBlock = connect(mapStateToProps, mapDispatchToProps)(InfoBlockUI);