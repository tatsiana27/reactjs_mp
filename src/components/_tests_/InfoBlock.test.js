import React from 'react';
import { InfoBlock } from '../InfoBlock';
import { shallow } from 'enzyme';

describe('InfoBlock ', () => {
    test('should render InfoBlock Component', () => {
        const wrapper = shallow(
            <InfoBlock/>

        );

        expect(wrapper).toMatchSnapshot();
    });
});