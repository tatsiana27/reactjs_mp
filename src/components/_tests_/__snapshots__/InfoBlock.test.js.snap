// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBlock  should render InfoBlock Component 1`] = `
<div
  className="info-block"
>
  <span
    className="movies-count"
  >
    0 movies found
  </span>
  <div
    className="sorting-block"
  >
    <p>
      Sort by
    </p>
    <RadioButton
      id={1}
      isChecked={false}
      key="1"
      name="release date"
      value="release date"
    />
    <RadioButton
      id={2}
      isChecked={true}
      key="2"
      name="raiting"
      value="raiting"
    />
  </div>
</div>
`;
